### PR TITLE
feat: use post script to complete deployment

### PR DESCRIPTION
**Pull request by radiantspace**
This implements a new feature and creates a breaking change:
- Instead of manually calling finish type action, use post script to complete those deployments automatically.

Relates to github/npm#530

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,9 +7,9 @@
   "scripts": {
     "build": "tsc",
     "lint": "ts-standard src/**/*.ts",
-    "pack": "ncc build",
+    "pack": "ncc build lib/main.js -o dist/main && ncc build lib/post.js -o dist/post",
```
Comment (radiantspace) [inline package.json:10]: build both entry points for main and post scripts

File: README.md [hunk 2]
```diff
@@ -17,9 +16,10 @@ Inputs:
 `token`|**Required** token to authorize calls to GitHub API, can be ${{github.token}} to create a deployment for the same repo
 `type`|**Required** type of an action. Should be `create` to create a deployment
 `logs`|url to the deployment logs
-`environment`|environment to create a deployments in, default to `context.ref` without prefixes (`'refs/heads/'`, `'deploy-'`), i.e. branch name
+`environment`|environment to create a deployments in, default to `$context.ref` without prefixes (`'refs/heads/'`, `'deploy-'`), i.e. branch name
 `environment_url`|link to the deployed application
 `description`|optional description, defaults to `"deployed by $context.actor"`
+`job_status`| pass `${{job.status}}` to set the deployment completion status post script accordingly
```
Comment (radiantspace) [inline README.md:22]: @gordey4doronin @lumaxis that's a dirty hack to track deployment status in post script, unless we find a way to query ${{job.status}} in JS code.

File: action.yml [hunk 3]
```diff
@@ -29,6 +29,10 @@ inputs:
   main_branch:
     required: false
     description: name of main branch of the repo, to get and store sha of the head commit and track diff being deployed
+  job_status:
```
Comment (lumaxis) [inline action.yml:32]: Do we actually need to pass this in? I saw that we have access to `github.event` in the `post` script so I'm wondering why it's different with `job`.

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -32,13 +33,7 @@ jobs:
         environment: test
         logs: https://github.com/npm/action-deploy/actions?query=workflow%3Abuild-test
         environment_url: https://npmjs.com
-    - uses: ./
-      name: Finish first deployment with success
-      with:
-        token: ${{github.token}}
-        type: finish
-        status: success
-        deployment_id: ${{steps.create-first.outputs.deployment_id}}
```
Comment (lumaxis) [inline .github/workflows/test.yml:41]: 👏🏼

File: action.yml (hunk 3 above)
Comment (gordey4doronin) [reply action.yml:32]: You can do

```js
import * as github from '@actions/github';

github.context.repo
```

But we were not able to find how to access either `event` or `job` 🤔

Comment (radiantspace) [reply action.yml:32]: unfortunately no job context is accessible in JS code - https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions, only in YAML. Only env context and github context.

Comment (radiantspace) [reply action.yml:32]: also we can use states (which are essentially just process.env vars with STATE_ prefix), but we need to know what happened to additional steps after our action.

Comment (lumaxis) [reply action.yml:32]: Okay, cool 👍🏼

I'm still wondering if there's a secret way to do it. Asked in #c2c-actions-support, we can always adjust in the future if there actually are other ways.

**Pull request merged**